Documentation updates

donnyh13 opened the pull request:
## Pull request

### Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.<br>
Please ensure you have read and noticed the checklist below.

### Checklist

Put an `x` in the boxes that apply. If you're unsure about any of them, don't hesitate to ask. We are here to help!<br>
This is simply a reminder of what we are going to look for before merging your code.

- [x] I have read and noticed the [CODE OF CONDUCT](https://github.com/mlipok/Au3LibreOffice/tree/main/Docs/CODE_OF_CONDUCT.md) document
- [x] I have read and noticed the [CONTRIBUTING](https://github.com/mlipok/Au3LibreOffice/tree/main/Docs/CONTRIBUTING.md) document
- [x] I have added necessary documentation or screenshots (if appropriate)
- [ ] I have tested my changes or additions for proper functionality.
- [x] I have followed, as best as possible, the same function layout and coding style of other existing functions/code.

### Types of changes

Please check `x` the type of change your PR introduces:

- [ ] Bugfix (change which fixes an issue)
- [ ] Feature (change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (functional, structural)
- [x] Documentation content changes
- [ ] Other (please describe) <br>
**	|-->:

### What is the current behavior?

Please describe the current behavior that you are modifying, or link to a relevant issue. <br>
**	|-->:

### What is the new behavior?

Please describe the behavior or changes that are being added by this PR. <br>
**	|-->:

### Influences and relationship to other functionality

Describe how the changes will affect other functions, potential script breaking changes, etc. <br>
**	|-->:

### Additional context

Add any other context about the problem here. <br>
**	|-->: Modify ReadMe to add detail. Modify au3 File header to describe what each does. Remove repetitive wording. 

### System under test

Please complete the following information.

- OS: [e.g. Windows 10]
- OS Arch.: [e.g. X64]
- LibreOffice Component [e.g. Main, Writer, Calc, ]
- LibreOffice version [e.g. 7.3.4.2]

donnyh13 committed:
Create gitattributes

donnyh13 committed:
Header Index re-wording.

donnyh13 committed:
Update README.md

donnyh13 committed:
Update CHANGELOG.md

donnyh13 committed:
Update README.md

donnyh13 committed:
Update README.md

donnyh13 committed:
Update README.md

donnyh13 committed:
Update README.md

donnyh13 committed:
Update README.md

mlipok commented on LibreOfficeWriter_Font.au3:
```diff
@@ -6,23 +6,12 @@
 ; AutoIt Version : v3.3.16.1
-; UDF Version    : 0.0.0.3
-; Description ...: Provides basic functionality through Autoit for interacting with Libre Office Writer.
+; Description ...: Provides basic functionality through AutoIt for Listing and querying available L.O. Writer Font Styles.
```
Font Styles ?

donnyh13 replied:
Sorry, should be "Fonts"

mlipok replied:
no.
My mistake. Sorry

mlipok commented on LibreOfficeWriter_Field.au3:
```diff
@@ -8,23 +8,12 @@
 ; AutoIt Version : v3.3.16.1
-; UDF Version    : 0.0.0.3
-; Description ...: Provides basic functionality through Autoit for interacting with Libre Office Writer.
+; Description ...: Provides basic functionality through AutoIt for Inserting and Modifying L.O. Writer Fields.
```
Fields ?

donnyh13 replied:
Yes, I mean like this:
https://help.libreoffice.org/latest/en-US/text/swriter/guide/fields.html

mlipok replied:
My mistake. I was looking wrong place.
![image](https://github.com/mlipok/Au3LibreOffice/assets/11089482/e7f1225b-30a2-41ab-9869-0bb8ebcfa6a5)

donnyh13 replied:
Oh Okay. I understand now.

mlipok commented on LibreOfficeWriter_Doc.au3:
```diff
@@ -8,23 +8,12 @@
 ; AutoIt Version : v3.3.16.1
-; UDF Version    : 0.0.0.3
-; Description ...: Provides basic functionality through Autoit for interacting with Libre Office Writer.
+; Description ...: Provides basic functionality through AutoIt for Creating, Modifying, Closing, Saving, Searching, etc. L.O. Writer documents.
```
documents ?

donnyh13 replied:
You would prefer singular? Or what are your thoughts?

mlipok replied:
Here was my mistake.
Sorry.

mlipok commented on LibreOfficeWriter_DirectFormating.au3:
```diff
@@ -10,23 +10,12 @@
 ; AutoIt Version : v3.3.16.1
-; UDF Version    : 0.0.0.3
-; Description ...: Provides basic functionality through Autoit for interacting with Libre Office Writer.
+; Description ...: Provides basic functionality through AutoIt for Modifying and Applying Direct Character and Paragraph formating in L.O. Writer.
```
Paragraph ?

donnyh13 replied:
Yes, some direct formatting is applied specifically for Characters, others, such as Tab Stops,  etc, I believe are applied specifically to paragraphs. At least as best as I can tell.

mlipok replied:
I was confused but now understand.
Thanks.

mlipok commented on LibreOfficeWriter_Char.au3:
```diff
@@ -8,23 +8,12 @@
 ; AutoIt Version : v3.3.16.1
-; UDF Version    : 0.0.0.3
-; Description ...: Provides basic functionality through Autoit for interacting with Libre Office Writer.
+; Description ...: Provides basic functionality through AutoIt for Creating, Modifying, and Applying L.O. Writer Character Styles.
```
styles ?

donnyh13 replied:
Yes, Libre Office has two ways of applying such things as "Bold", Direct formatting and Character Styles.
https://help.libreoffice.org/7.2/en-US/text/swriter/01/05130002.html

mlipok replied:
but here we have LibreOfficeWriter_Char.au3 not styles.... so I was confused.
But now I get it what you mean.
`Characters Styles` not `Styles`

mlipok commented on LibreOfficeWriter_FootEndNotes.au3:
```diff
@@ -6,23 +6,12 @@
 ; AutoIt Version : v3.3.16.1
-; UDF Version    : 0.0.0.3
-; Description ...: Provides basic functionality through Autoit for interacting with Libre Office Writer.
+; Description ...: Provides basic functionality through AutoIt for Creating, Modifying, and Inserting L.O. Writer Footnotes and Endnotes.
```
Foot/End notes ?

donnyh13 replied:
Okay

mlipok commented on LibreOfficeWriter_Frame.au3:
```diff
@@ -8,23 +8,12 @@
 ; AutoIt Version : v3.3.16.1
-; UDF Version    : 0.0.0.3
-; Description ...: Provides basic functionality through Autoit for interacting with Libre Office Writer.
+; Description ...: Provides basic functionality through AutoIt for Creating, Modifying, and Applying L.O. Writer Frame Styles. Also for Creating, and Modifying Frames.
```
Frames ?

donnyh13 replied:
Yes this file deals with both Frames and Frame Styles.

mlipok replied:
my mistake. Sorry.

mlipok commented on LibreOfficeWriter_Helper.au3:
```diff
@@ -7,23 +7,12 @@
 ; AutoIt Version : v3.3.16.1
-; UDF Version    : 0.0.0.3
-; Description ...: Provides basic functionality through Autoit for interacting with Libre Office Writer.
+; Description ...: Functions used for creating, modifying and retrieving data for use in various functions in LibreOffice UDF.
```
should be Page ?

donnyh13 replied:
What do you mean?

mlipok replied:
My mistake.
Sorry.

mlipok commented on LibreOfficeWriter_Images.au3:
```diff
@@ -9,23 +9,12 @@
 ; AutoIt Version : v3.3.16.1
-; UDF Version    : 0.0.0.3
-; Description ...: Provides basic functionality through Autoit for interacting with Libre Office Writer.
+; Description ...: Provides basic functionality through AutoIt for Inserting and Modifying Images in L.O. Writer.
```
Images ?

mlipok replied:
My mistake sorry.

mlipok commented on LibreOfficeWriter_Internal.au3:
```diff
@@ -7,23 +7,12 @@
 ; AutoIt Version : v3.3.16.1
-; UDF Version    : 0.0.0.3
-; Description ...: Provides basic functionality through Autoit for interacting with Libre Office Writer.
+; Description ...: Various functions for internal data processing, data retrieval, data processing, retrieving and applying settings for LibreOffice UDF.
```
ok

mlipok commented on LibreOfficeWriter_Num.au3:
```diff
@@ -8,23 +8,12 @@
 ; AutoIt Version : v3.3.16.1
-; UDF Version    : 0.0.0.3
-; Description ...: Provides basic functionality through Autoit for interacting with Libre Office Writer.
+; Description ...: Provides basic functionality through AutoIt for Creating, Modifying, and Applying L.O. Writer Numbering Styles.
```
should be char ?

mlipok replied:
My mistake. Sorry.

mlipok commented on LibreOfficeWriter_Page.au3:
```diff
@@ -8,23 +8,12 @@
 ; AutoIt Version : v3.3.16.1
-; UDF Version    : 0.0.0.3
-; Description ...: Provides basic functionality through Autoit for interacting with Libre Office Writer.
+; Description ...: Provides basic functionality through AutoIt for Creating, Modifying, and Applying L.O. Writer Page Styles.
```
Page ? should be Par ?

mlipok replied:
My mistake. sorry.

mlipok commented on LibreOfficeWriter_Par.au3:
```diff
@@ -11,23 +11,12 @@
 ; AutoIt Version : v3.3.16.1
-; UDF Version    : 0.0.0.3
-; Description ...: Provides basic functionality through Autoit for interacting with Libre Office Writer.
+; Description ...: Provides basic functionality through AutoIt for Creating, Modifying, and Applying L.O. Writer Paragraph Styles.
```
Paragraph ? or should be Page ?

mlipok replied:
My mistake. Sorry.

mlipok commented on LibreOfficeWriter_Shapes.au3:
```diff
@@ -5,23 +5,12 @@
 ; AutoIt Version : v3.3.16.1
-; UDF Version    : 0.0.0.3
-; Description ...: Provides basic functionality through Autoit for interacting with Libre Office Writer.
+; Description ...: Provides basic functionality through AutoIt for Creating, Modifying, and Inserting shapes in L.O. Writer.
```
Ok.

mlipok commented on LibreOfficeWriter_Table.au3:
```diff
@@ -8,23 +8,12 @@
 ; AutoIt Version : v3.3.16.1
-; UDF Version    : 0.0.0.3
-; Description ...: Provides basic functionality through Autoit for interacting with Libre Office Writer.
+; Description ...: Provides basic functionality through AutoIt for Creating, Modifying, and Inserting Tables in L.O. Writer.
```
Tables ? or Page ?

mlipok replied:
My mistake.

donnyh13 committed:
Typo correction

mlipok commented:
It appears that in many cases you have confused the description you provided with the file in question.

I hope you can fix it soon.

donnyh13 committed:
Update LibreOfficeWriter_Font.au3

donnyh13 committed:
Update LibreOfficeWriter_FootEndNotes.au3

donnyh13 commented:
> It appears that in many cases you have confused the description you provided with the file in question.
> 
> I hope you can fix it soon.

Sorry just saw this message, have been replying to your individual comments.
 I'll review, again. 
Edit*
Unless there is some strange glitch between my computer and yours, most of the changes are looking correct on my end????

For example the file LibreOfficeWriter_Page.au3 shows these lines:
Removed ~~; Description ...: Provides basic functionality through Autoit for interacting with Libre Office Writer.~~
Added **; Description ...: Provides basic functionality through AutoIt for Creating, Modifying, and Applying L.O. Writer Page Styles.**

And your comment is:
> Page ? should be Par ?

mlipok commented:
I was looking in wrong place:
![image](https://github.com/mlipok/Au3LibreOffice/assets/11089482/014eeb6f-0bcd-4a9f-bef7-a29f33b5da36)

My mistake. Sorry for confusion.

donnyh13 commented:
No problem. Thanks for clarifying.
